pkg/controller/servicebindingrequest/binder.go

```diff
@@ -501,14 +513,23 @@ func (b *binder) update(objs *unstructured.UnstructuredList) ([]*unstructured.Un
 
 		sbrNamespacedName := types.NamespacedName{Namespace: b.sbr.GetNamespace(), Name: b.sbr.GetName()}
 		updatedObj = setSBRAnnotations(sbrNamespacedName, updatedObj)
+		var err error
+		if b.sbr.Spec.ApplicationSelector.BindingPath.SecretPath != "" {
+			err = b.updateSecretField(updatedObj)
+			if err != nil {
+				return nil, err
+			}
+		}
 
-		updatedObj, err := b.updateSpecContainers(updatedObj)
-		if err != nil {
-			return nil, err
+		if b.sbr.Spec.ApplicationSelector.BindingPath.ContainersPath != "" {
```

Comment:
I presume `ensureDefaults` should take care of this, shouldn't it? If not, having an empty string `""` as valid value contradicts having a default for `containersPath`.

Unless this is also changing semantics where not providing a `ContainersPath` indicates that no related container paths changes will be attempt in the application resource?

Reply:
my understanding here is that with this here we could do the following:

```yaml
bindingPath:
  containerPath: ''
  secretPath: spec.foo.bar
```

With such config, we could fully skip injecting any env vars, and add only secretRef to the given path. I can see the usecase that.

Reply:
@pedjak 

I agree with you that this is indeed a good use case, but I believe that breaking the current behavior (which this is indeed doing) is not what we want in the scope of this PR, wouldn't you agree?

Reply:
I have made changes to set default path for container path when the secret path is specified without container path.

Reply:
I hope this scenario is going to help to understand this issue better: https://github.com/redhat-developer/service-binding-operator/blob/6d1a8034abbe7a7a4e3f9bbc73353aa3112ee686/test/acceptance/features/injectSecretAtCustomSchemaPath.feature#L122

Reply:
> I agree with you that this is indeed a good use case, but I believe that breaking the current behavior (which this is indeed doing) is not what we want in the scope of this PR, wouldn't you agree?

@isutton 

How do we break the current behaviour, since we do not have possibility to configure bindingPath on master?

This PR introduces ability to specify custom binding path. Forcing users to have appropriate container path in app CR reduces the flexibility of the whole approach and its usability.

We had this discussion recently:

https://github.com/redhat-developer/service-binding-operator/pull/513#discussion_r469314671

and came to the agreement. Let's stay on this.

Reply:
I have made the necessary changes. Also squashed the commits.

Reply:
> > I agree with you that this is indeed a good use case, but I believe that breaking the current behavior (which this is indeed doing) is not what we want in the scope of this PR, wouldn't you agree?
> 
> @isutton
> 
> How do we break the current behaviour, since we do not have possibility to configure bindingPath on master?
> 
> This PR introduces ability to specify custom binding path. Forcing users to have appropriate container path in app CR reduces the flexibility of the whole approach and its usability.
> 
> We had this discussion recently:
> 
> [#513 (comment)](https://github.com/redhat-developer/service-binding-operator/pull/513#discussion_r469314671)
> 
> and came to the agreement. Let's stay on this.

PodSpec patching will happen:

1) in the case `bindingPath` is missing entirely; or
2) in the case `bindingPath` is defined and `containersPath` contains a value.

PodSpec patching will not happen:

1) in the case `bindingPath` is defined and `containersPath` contains an empty value.

If the user is not declaring `bindingPath` but starts to (by defining `bindingPath.secretPath`), PodSpec should cease be patched at the next reconciliation.

Isn't adding `bindingPath.secretPath` disabling the PodSpec patch? Is in adding `bindingPath.secretPath` the user's intention to disable PodSpec patching or not? 

I particularly think the aforementioned action totally is different than explicitly specifying `bindingPath.containersPath` to `""`, which the user is indicating the action should not be performed.

Regardless of what I think, if we think this is the correct behavior this should be properly present in the documentation.